### PR TITLE
feat: Pause previous deliveries when the test taker launches a new one

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/TR-5852

### CODE REVIEW

File: controller/DeliveryTool.php [hunk 1]
```diff
@@ -307,9 +318,14 @@ protected function getTool()
...
     {
+        /** @var Ontology $ontology */
+        $ontology = $this->getServiceManager()->get(Ontology::SERVICE_ID);
+
         //passed as a parameter
         if ($this->hasRequestParameter('delivery')) {
-            $returnValue = new core_kernel_classes_Resource($this->getRequestParameter('delivery'));
+            $returnValue = $ontology->getResource($this->getRequestParameter('delivery'));
+
+            //$returnValue = new core_kernel_classes_Resource($this->getRequestParameter('delivery'));
```
Comment (gabrielfs7) [inline controller/DeliveryTool.php:328]: - [ ] We can revert this

File: controller/DeliveryTool.php [hunk 2]
```diff
@@ -103,6 +104,16 @@ class DeliveryTool extends ToolModule
     public function run()
     {
         $compiledDelivery = $this->getDelivery();
+
+        //FIXME
+        //FIXME
+        // @TODO Pause the previous delivery for the same user in case it is active
```
Comment (gabrielfs7) [inline controller/DeliveryTool.php:110]: - [ ] Probably the place we need to work to pause previous delivery and timer

File: composer.json [hunk 3]
```diff
@@ -68,7 +68,7 @@
 		"oat-sa/extension-tao-delivery-rdf" : ">=14.0.0",
 		"oat-sa/extension-tao-lti" : ">=15.11.0",
 		"oat-sa/extension-tao-outcomeui" : ">=10.0.0",
-		"oat-sa/extension-tao-testqti" : ">=42.2.0",
+		"oat-sa/extension-tao-testqti" : "dev-feat/TR-5852/disable-multiple-delivery-launch as >=42.2.0",
```
Comment (hectoras) [inline composer.json:71]: - [ ] Update with released version

File: controller/DeliveryTool.php (hunk 1 above)
Comment (hectoras) [reply controller/DeliveryTool.php]: Done

File: controller/DeliveryTool.php [hunk 4]
```diff
@@ -121,48 +140,189 @@ public function run()
...
+        }
+    }
 
-            $isDryRun = !$isLearner && in_array(LtiRoles::CONTEXT_LTI1P3_INSTRUCTOR, $user->getRoles(), true);
+    protected function runTestSession(
+        common_session_Session $session,
+        core_kernel_classes_Resource $compiledDelivery
+    ): void {
+        $user = $session->getUser();
+        $ltiRoles = [LtiRoles::CONTEXT_LEARNER, LtiRoles::CONTEXT_LTI1P3_LEARNER];
```
Comment (hectoras) [inline controller/DeliveryTool.php:152]: TODO
- [ ] May be moved into a class constant

**Comment by hectoras** [comment]
Superseded by https://github.com/oat-sa/extension-tao-ltideliveryprovider/pull/365

**Pull request closed without merging**